chat-plugins/datasearch.js

```diff
@@ -85,11 +85,19 @@ const PM = exports.PM = new DatasearchManager({
 exports.commands = {
 	'!dexsearch': true,
 	ds: 'dexsearch',
+	ds1: 'dexsearch',
+	ds2: 'dexsearch',
+	ds3: 'dexsearch',
+	ds4: 'dexsearch',
+	ds5: 'dexsearch',
+	ds6: 'dexsearch',
+	ds7: 'dexsearch',
 	dsearch: 'dexsearch',
 	dexsearch: function (target, room, user, connection, cmd, message) {
 		if (!this.canBroadcast()) return;
+		let targetGen = parseInt(cmd[cmd.length - 1]);
+		if (targetGen) target += ", gen<" + (targetGen + 1);
```

Comment:
This doesn't just apply to, like, gen range, you know. It also affects learnsets, stats, abilities...